.github/workflows/sonar.yaml

```
@@ -24,5 +24,5 @@ jobs:
       - name: Upload coverage reports to Sonar
         env:
           SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
-        if: github.event.pull_request.head.repo.full_name == github.repository || env.SONAR_TOKEN != ''
-        uses: SonarSource/sonarqube-scan-action@v5.3.0
+        if: (github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name == github.repository) || (github.event_name != 'pull_request' && secrets.SONAR_TOKEN != '')
```

actionlint (1.7.7)

context "secrets" is not allowed here. available contexts are "env", "github", "inputs", "job", "matrix", "needs", "runner", "steps", "strategy", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)